java/rocksjni/ratelimiterjni.cc

```diff
@@ -22,3 +23,98 @@ jlong Java_org_rocksdb_GenericRateLimiterConfig_newRateLimiterHandle(
       static_cast<int64_t>(jrefill_period_micros),
       static_cast<int32_t>(jfairness)));
 }
+
+/*
+ * Class:     org_rocksdb_RateLimiter
+ * Method:    newRateLimiterHandle
+ * Signature: (JJI)J
+ */
+jlong Java_org_rocksdb_RateLimiter_newRateLimiterHandle(
+    JNIEnv* env, jclass jclazz, jlong jrate_bytes_per_second,
+    jlong jrefill_period_micros, jint jfairness) {
+  auto* rate_limiter = rocksdb::NewGenericRateLimiter(
+      static_cast<int64_t>(jrate_bytes_per_second),
+      static_cast<int64_t>(jrefill_period_micros),
+      static_cast<int32_t>(jfairness));
+
+  // TODO(AR) `ptr_sptr_rate_limiter` is never deleted :-(
+  std::shared_ptr<rocksdb::RateLimiter> *ptr_sptr_rate_limiter =
+    new std::shared_ptr<rocksdb::RateLimiter>;
+  *ptr_sptr_rate_limiter = std::shared_ptr<rocksdb::RateLimiter>(rate_limiter);
+
+  return reinterpret_cast<jlong>(ptr_sptr_rate_limiter);
+}
+
+/*
+ * Class:     org_rocksdb_RateLimiter
+ * Method:    disposeInternal
+ * Signature: (J)V
+ */
+void Java_org_rocksdb_RateLimiter_disposeInternal(
+    JNIEnv* env, jobject jobj, jlong jhandle) {
+  // TODO(AR) `ptr_sptr_rate_limiter` is never deleted :-(
+  std::shared_ptr<rocksdb::RateLimiter> *handle =
+      reinterpret_cast<std::shared_ptr<rocksdb::RateLimiter> *>(jhandle);
+  handle->reset();
```

Comment:
I don't think so, there could be more than one Java object which uses the same native object as that is the point of a shared_ptr I think... Also if I delete the underlying memory of the Native Object object, how do I know that the database still doesn't need access to it?